examples/model_selection/plot_grid_search_refit_callable.py

```diff
@@ -79,49 +101,293 @@ def best_low_complexity(cv_results):
+        alpha=0.5,
+        color="blue",
+        label="Test scores" if i == 0 else "",
+    )
+
+# Add mean lines
+ax2.plot(
+    n_components,
+    np.mean(train_scores, axis=1),
+    "-",
+    color="#ffd700",  # Yellow for train scores
+    linewidth=2,
+    label="Mean train score",
+)
+ax2.plot(
+    n_components,
+    np.mean(test_scores, axis=1),
+    "-",
+    color="blue",
+    linewidth=2,
+    label="Mean test score",
+)
+
+# Highlight selected model in second plot
+ax2.axvline(
+    best_components,
+    color="#9b59b6",  # Purple
+    alpha=0.2,
+    linewidth=8,
+    label="Selected model",
+)
+
+# Set titles and labels for both subplots
+for ax in (ax1, ax2):
+    ax.set_xlabel("Number of PCA components", fontsize=12)
+    ax.set_xticks(n_components)
+    ax.set_ylim((0.7, 1.0))
+    ax.grid(True, linestyle="--", alpha=0.7)
+    ax.legend(loc="lower right")
+
+ax1.set_title("Model Selection Criteria", fontsize=14)
+ax1.set_ylabel("Test accuracy", fontsize=12)
+
+ax2.set_title("Train vs. Test Scores", fontsize=14)
+ax2.set_ylabel("Score", fontsize=12)
+
+# Add a main title for the entire figure
+fig.suptitle("Balance model complexity and cross-validated score", fontsize=16, y=1.05)
+
+# %%
+# Print the results
+# ---------------
+#
+# We print information about the selected model, including its complexity and
+# performance. We also show a summary table of all models using polars.
+
+print("Best model selected by the one-standard-error rule:")
+print(f"Number of PCA components: {best_components}")
+print(f"Accuracy score: {best_score:.4f}")
+print(f"Best possible accuracy: {np.max(test_scores):.4f}")
+print(f"Accuracy threshold (best - 1 std): {lower:.4f}")
+
+# Create a summary table with polars
+summary_df = results_df.select(
+    pl.col("n_components"),
+    pl.col("mean_test_score").round(4).alias("test_score"),
+    pl.col("std_test_score").round(4).alias("test_std"),
+    pl.col("mean_train_score").round(4).alias("train_score"),
+    pl.col("std_train_score").round(4).alias("train_std"),
+    pl.col("mean_fit_time").round(3).alias("fit_time"),
+    pl.col("rank_test_score").alias("rank"),
+)
+
+# Add a column to mark the selected model
+summary_df = summary_df.with_columns(
+    pl.when(pl.col("n_components") == best_components)
+    .then(pl.lit("*"))
+    .otherwise(pl.lit(""))
+    .alias("selected")
+)
+
+print("\nModel comparison table:")
+print(summary_df)
+
+# %%
+# Conclusion
+# ---------
+#
+# The one-standard-error rule helps us select a simpler model (fewer PCA components)
+# while maintaining performance statistically comparable to the best model.
+# This approach can help prevent overfitting and improve model interpretability
+# and efficiency.
+#
+# In this example, we've seen how to implement this rule using a custom refit
+# callable with :class:`~sklearn.model_selectoin.GridSearchCV`.
+#
+# Key takeaways:
+# 1. The one-standard-error rule provides a principled way to select simpler models
```

Comment:
Non actionable note: it is no principle, just a rule of thumb without any theoretical foundation.

NB: Sometimes, those rules turn out to be the best ones 🤣

Reply:
I would vote to change "a principled way to" to a "a pragmatic rule of thumb to".